include/ebpf_structs.h

```diff
@@ -77,3 +77,31 @@ typedef enum
     BPF_FUNC_get_smp_processor_id = 7,
     BPF_FUNC_ktime_get_ns = 8,
 } ebpf_helper_id_t;
+
+// Libbpf itself requires the following structs to be defined, but doesn't
+// care what fields they have.  Applications such as bpftool on the other
+// hand depend on fields of specific names and types.
+
+struct bpf_link_info
```

Comment:
I think bpf_link_info should also have `attach_type`. bpf_link_info in Linux also contains attach type (though inside union of structs). The API doc also mentions `attach_type` in ebpf_link_info

Reply:
there is no bpf_link_info.attach_type field, there is only attach_type down another level inside structs we don't support yet like bpf_link_info.cgroup.attach_type, bpf_link_info.tracing.attach_type, etc.  This PR only adds standard fields that we support.